src/Compiler/Microsoft.NET.Sdk.Razor.SourceGenerators/RazorSourceGenerator.RazorProviders.cs

```diff
@@ -70,7 +70,8 @@ private static (SourceGeneratorProjectItem?, Diagnostic?) ComputeProjectItems((A
             var (additionalText, globalOptions) = pair;
             var options = globalOptions.GetOptions(additionalText);
 
-            if (!options.TryGetValue("build_metadata.AdditionalFiles.TargetPath", out var encodedRelativePath))
+            if (!options.TryGetValue("build_metadata.AdditionalFiles.TargetPath", out var encodedRelativePath) ||
+                string.IsNullOrEmpty(encodedRelativePath))
```

Comment:
Consider doing `string.IsNullOrWhitespace`, and adding a test with only whitespace in the target path.

Reply:
Sounds good, thanks. Done.